walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,16 +27,63 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    List<TimeRange> freeTimes = new ArrayList<>();
+    // Check for edge cases.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return freeTimes;
+    }
+    if (events.isEmpty() || request.getAttendees().isEmpty()) {
+      freeTimes.add(TimeRange.WHOLE_DAY);
+      return freeTimes;
+    }
```

Comment:
Dealing with this edge case here isn't really necessary because it will be caught in lines 55-57 but I'm wondering if it's better to catch this up front anyway?

Reply:
Yeah I think it's better to catch up front so that we don't have to go through unnecessary code

Reply:
But might be best to add some sort of comment in findFreeIntervals stating that busyTimes should not be empty, and make sure you have a test that checks if there are no attendees than the free times returns the whole day

Reply:
Yep optionsForNoAttendees does that. And I think even with this check busyTimes still could be empty because we could have a list of events with no relevant attendees (so none of them would get added to busyTimes)